src/NServiceBus.Core/Pipeline/RootContext.cs

```diff
@@ -1,16 +1,12 @@
 namespace NServiceBus
 {
-    using MessageInterfaces;
     using ObjectBuilder;
 
     class RootContext : BehaviorContext
     {
-        public RootContext(IBuilder builder, IPipelineCache pipelineCache, IEventAggregator eventAggregator, IMessageMapper messageMapper) : base(null)
+        public RootContext(IBuilder builder) : base(null)
```

Comment:
it would be nice if we could somehow prevent others than the PipelineComponent to create a RootContext given the component has the only knowledge of what exactly needs to be set on the root context.